lib/charms/traefik_route_k8s/v0/traefik_route.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+r"""# Interface Library for traefik_route.
+
+This library wraps relation endpoints for traefik_route. The requirer of this
+relation is a charm in need of ingress (or a proxy thereof), the
+provider is the traefik-k8s charm.
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.traefik_route_k8s.v0.traefik_route
+```
+
+```yaml
+requires:
+    traefik_route:
+        interface: traefik_route
+        limit: 1
+```
+
+Then, to initialise the library:
+
+```python
+# ...
+from charms.traefik_route_k8s.v0.traefik_route import TraefikRouteRequirer
+
+class SomeCharm(CharmBase):
+  def __init__(self, *args):
+    # ...
+    self.ingress_per_unit = TraefikRouteProvider(self)
+    self.traefik_route = TraefikRouteRequirer(self)
+
+    self.framework.observe(
+        self.ingress_per_unit.on.request, self.traefik_route.relay
+    )
+    self.framework.observe(
+        self.traefik_route.on.response, self.ingress_per_unit.respond
+    )
```

Comment:
As above - I think something was lost in translation here.

Reply:
Uh yes, copy-pasted from the IPU library a bit too freely. On it